DBAL/Types/AbstractEnumType.php

```diff
@@ -192,7 +198,7 @@ public static function getReadableValues(): array
     /**
      * Asserts that given choice exists in the array of ENUM values.
      *
-     * @param mixed $value ENUM value
+     * @param T $value ENUM value
```

Comment:
This is quite cool because with this the user will be alerted if they provide a value that's not supported by their enum based on the type at all, e.g.

```PHP
BasketballPositionType::assertValidChoice(5.0)

Reply:
sadly after testing with Psalm analysis at a stricter level I realised this doesn't work because they are static methods and the template T cannot be linked to the T on the static method. I will revert some of these

Reply:
fixed those, it was just me not getting the types right at first